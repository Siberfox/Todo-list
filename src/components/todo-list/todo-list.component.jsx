import React from "react";
import { TodoItem } from "../todo-item/todo-item.component";

import "./todo-list.styles.scss";

export const TodoList = ({ items }) => {
  return (
    <ul className="todo-list">
      <h1 className="title">TODO LIST</h1>
      {items.map((item) => (
        <TodoItem item={item} />
      ))}
    </ul>
  );
};
