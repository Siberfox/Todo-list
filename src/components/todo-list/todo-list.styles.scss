.todo-item {
  margin-bottom: 20px;
  text-align: left;
  word-wrap: break-word;
}
.todo-list {
  padding: 0;
}
