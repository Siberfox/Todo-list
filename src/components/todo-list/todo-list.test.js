import React from "react";
import { shallow } from "enzyme";

import { TodoList } from "./todo-list.component";

describe("Todo-list", () => {
  const mockItems = [
    { text: "first", completed: false, id: 1 },
    { text: "second", completed: false, id: 2 },
  ];
  const mockRemoveItem = jest.fn();
  const mockHandleClick = jest.fn();

  const wrapper = shallow(
    <TodoList
      items={mockItems}
      removeItem={mockRemoveItem}
      handleClick={mockHandleClick}
    />
  );

  it("should render todo-list component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("component have list from items props", () => {
    expect(wrapper.find("ul").children()).toHaveLength(2);
  });

  it("should call handleClick on click", () => {
    wrapper.find(".todo-item_text").at(1).simulate("click");
    expect(mockHandleClick).toHaveBeenCalled();
  });

  it("should remove item on click", () => {
    wrapper.find(".trash-icon").at(1).simulate("click");
    expect(mockRemoveItem).toHaveBeenCalled();
  });
});
