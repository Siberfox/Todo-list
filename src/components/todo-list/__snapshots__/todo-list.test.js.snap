// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo-list should render todo-list component 1`] = `
<ul
  className="todo-list"
>
  <li
    className="todo-item"
    key="1"
  >
    <span
      className="todo-item_text "
      onClick={[Function]}
    >
      first
    </span>
    <ForwardRef(SvgTrash)
      className="trash-icon"
      onClick={[Function]}
    />
  </li>
  <li
    className="todo-item"
    key="2"
  >
    <span
      className="todo-item_text "
      onClick={[Function]}
    >
      second
    </span>
    <ForwardRef(SvgTrash)
      className="trash-icon"
      onClick={[Function]}
    />
  </li>
</ul>
`;
